src/dishka/registry.py

```diff
@@ -345,11 +346,11 @@ def _process_decorator(
     def _process_context_var(
             self, provider: BaseProvider, context_var: ContextVariable,
     ) -> None:
-        registry = self.registries[context_var.scope]
+        registry = self.registries[cast(BaseScope, context_var.scope)]
```

Comment:
why cast?

Reply:
self.registries is typed as Dict[BaseScope, Registry] and factory.scope is of type BaseScope | None

Reply:
Looks like we need validate if it None here.